src/math/trig.c

```diff
@@ -94,9 +97,197 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
 		xtmp = *xn >> (b_idx + 1);
 		ytmp = *b_yn >> (b_idx + 1);
 	}
-	/* Q2.30 format */
+	/* Q2.30 format -sine, cosine*/
 	*th_cdc_fxp = th_rad_fxp;
+}
+
+/**
+ * \CORDIC-based approximation for inverse cosine
+ * Arguments	: int32_t cosvalue
+ *		  int16_t numiters
+ *		  const int32_t LUT[CORDIC_30B_ITABLE_SIZE]
+ * Return Type	: int32_t
+ */
+int32_t is_scalar_cordic_acos(int32_t cosvalue, int16_t numiters,
+			      const int32_t *LUT)
+{
+	int32_t b_cosvalue;
+	int32_t b_i;
+	int32_t i;
+	int32_t x = 0;
+	int32_t xdshift;
+	int32_t xshift;
+	int32_t y = 0;
+	int32_t ydshift;
+	int32_t yshift;
+	int32_t z = 0;
+	int32_t sign;
+	int16_t j;
+	int16_t k;
 
+	b_cosvalue = (cosvalue >> 1);
+	/* Initialize the variables for the cordic iteration
+	 * angles less than pi/4, we initialize (x,y) along the x-axis.
+	 * angles greater than or equal to pi/4, we initialize (x,y)
+	 * along the y-axis. This improves the accuracy of the algorithm
+	 * near the edge of the domain of convergence
+	 */
+	if (b_cosvalue < cord_arcsincos_q28fl) {
+		x = 0;
+		y = cord_arcsincos_q30fl;
+		z = PI_DIV2_Q3_29;
+	} else {
+		x = cord_arcsincos_q30fl;
+		y = 0;
+		z = 0;
+	}
+
+	b_cosvalue <<= 1;
+	/* DCORDIC(Double CORDIC) algorithm */
+	/* Double iterations method consists in the fact that unlike the classical */
+	/* CORDIC method,where the iteration step value changes EVERY time, i.e. on */
+	/* each iteration, in the double iteration method, the iteration step value */
+	/* is repeated twice and changes only through one iteration */
+	i = numiters - 1;
+	for (b_i = 0; b_i < i; b_i++) {
+		j = (b_i + 1) << 1;
+		if (j >= 31)
+			j = 31;
+
+		if (b_i < 31)
+			k = (int16_t)b_i;
+		else
+			k = 31;
```

Comment:
I do think line#153 to line#160 could be more simplified.
j = 0
for (b_i = 0; b_i < i; b_i++) {
  if (j < 31)
    j += 2;
  else
    j = 31;

And I think (b_i < 31) and k are unnecessary, because numiters is given internally, we could by ourselves make sure it won't exceed 31.
Although these are tiny change, I do think the reduction of logics in the for-loop is healthy for computation speed :)

Reply:
If we don't take ` k `then xshift will be impacted 
k is minimum of `iteration number -1 ` and `maxshift`
`y = (y - ydshift) + sign * xshift;`
To shorted the loop timing, LUT is shorten in size controlled by `numiters`

Reply:
If we could make sure that b_i would never exceed 31, then literally k = b_i
b_i will not exceed numiters. I browse through the whole code the maximal numiters is CORDIC_31B_TABLE_SIZE, which is 31. Unless we have the plan to develop more fine-grain function (I don't think so...)

Reply:
j exceeds 31 (` j = 32` ) when `b_i `is 15, so cap `j = 31`. This happens at `i = 30` iteration number

Reply:
I'm saying "k" in last reply.
As for "j", my proposal is also provided above (but I found it has a bug, I have modified a bit):
```
j = 0;
for (b_i = 0; b_i < i; b_i++) {
  if (b_i < 15)
    j += 2;
  else
    j = 31;
```

But I think it could just save shift operations, and add operations after b_i >= 15. It should be insignificant.
I'm OK for both ways :)